internal/clientconn/route.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientconn
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/AlekSi/pointer"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// Route routes the message.
+//
+// Specific message handlers should do one of the following:
+//  * return normal response body;
+//  * return protocol error (*common.Error, possibly wrapped) - it will be returned to the client;
+//  * return any other error - it will be returned to the client as InternalError before terminating connection.
+//
+// They should not panic on bad input, but may do so in "impossible" cases.
+// They also should not use recover(). That allows us to use fuzzing.
+// (Panics terminate the connection without a response on a different level.)
+func (c *conn) route(ctx context.Context, reqHeader *wire.MsgHeader, reqBody wire.MsgBody,
```

Comment:
Not sure why we move that method away from all other `conn` methods. I think it belongs in `conn.go`, where the type is defined

Reply:
moved.